src/xrEngine/xr_object_list.cpp

```diff
@@ -492,66 +487,62 @@ IGameObject* CObjectList::Create(LPCSTR name)
     return O;
 }
 
-void CObjectList::Destroy(IGameObject* O)
+void CObjectList::Destroy(IGameObject* game_obj)
 {
-    if (0 == O)
+    if (nullptr == game_obj)
         return;
-    net_Unregister(O);
+    net_Unregister(game_obj);
 
     if (!Device.Paused())
     {
-        if (!m_crows[1].empty())
+        // if a game is paused list of other crows should be empty - Why?
+        if (!m_secondary_crows.empty())
         {
-            Msg("assertion !m_crows[1].empty() failed: %d", m_crows[1].size());
-
-            Objects::const_iterator i = m_crows[1].begin();
-            Objects::const_iterator const e = m_crows[1].end();
-            for (u32 j = 0; i != e; ++i, ++j)
-                Msg("%d %s", j, (*i)->cName().c_str());
-            VERIFY(Device.Paused() || m_crows[1].empty());
-            m_crows[1].clear();
+            Msg("assertion !m_other_crows.empty() failed: %d", m_secondary_crows.size());
+
+            u32 j = 0;
+            for (auto iter: m_secondary_crows)
```

Comment:
When replacing iterators with range-based for consider to use references. Otherwise `auto` will copy value and we don't wan't that, because this can lead to incorrect behavior in worst case.
```suggestion
            for (auto& iter : m_secondary_crows)
```

Reply:
Here `iter` is just a pointer to `IGameObject`, no need to use reference for pointer. Value copying is intended here. No incorrect behavior at all and it is more efficient.

Reply:
Hmm...